MBBSEmu/AppSettingsManager.cs

```diff
@@ -0,0 +1,343 @@
+using Microsoft.Extensions.Configuration;
+using NLog;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text.Json;
+
+namespace MBBSEmu
+{
+    /// <summary>
+    ///     Handles Application Settings and Configuration for MBBSEmu
+    ///
+    ///     Configuration Values are loaded from the specified appsettings.json file (command line) or the default
+    /// </summary>
+    public class AppSettingsManager
+    {
+        public readonly IConfigurationRoot ConfigurationRoot;
+
+        private readonly ILogger _logger;
+
+        /// <summary>
+        ///     Safe loading of appsettings.json for Configuration Builder
+        /// </summary>
+        /// <returns></returns>
+        public AppSettingsManager(ILogger logger)
+        {
+            _logger = logger;
+
+            if (!File.Exists(ConfigurationFileName))
+                throw new FileNotFoundException($"Unable to locate [{ConfigurationFileName}] emulator settings file.");
+
+            if (!IsValidJson(File.ReadAllText(ConfigurationFileName)))
+                throw new InvalidDataException($"Invalid JSON detected in [{ConfigurationFileName}]. Please verify the format & contents of the file are valid JSON.");
+
+            ConfigurationRoot = new ConfigurationBuilder().SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile(ConfigurationFileName, false, true)
+                .Build();
+        }
+
+        /// <summary>
+        ///     Configuration File being used by MBBSEmu
+        /// </summary>
+        public string ConfigurationFileName => Program._settingsFileName ?? Program.DefaultEmuSettingsFilename;
+
+        //Validate Config File
+        public string BBSTitle => GetBBSTitleSettings("BBS.Title");
```

Comment:
we setup all those mappings in `AppSettings.cs` but they're not reused here at all which leads to duplication and makes me sad

Reply:
There might not be a way around this one. We handle the config file once loaded through the dotnet Configuration handler via `Microsoft.Extensions.Configuration`, but needed the new class to serialize/validate the JSON generated by the new Wizard. We could probably refactor the use of the configuration runtime out in a future PR, since we abstract it away anyway.